### DISCUSSION TITLE
how to work with async and import

### DISCUSSION THREAD

**Discussion by kangaroolab** [borkdude/scittle#98] [General]
hi, I have following javascript code, which I like to convert to clojurescript that can run with Scittle...can you please advise how to convert it?

 ```
const script = this.shadowRoot.querySelector('script');
                
                if (script) {
                    (async () => {
                        const module = await import(URL.createObjectURL(new Blob([script.textContent], { type: 'text/javascript' })));
                        module.default(this.shadowRoot);
                    })();
                }
```

**Comment by borkdude** [comment]
@kangaroolab You can do the following. There is no such thing as `js-await` in scittle (there is in squint and cherry) so you just need to use promise interop.

```
(when script
  (-> (js/import ....) (.then (fn [module] ((.-default module) ...))))
```

**Reply by kangaroolab** [reply]
thanks but it doesn't seem supporting js/import as I got error:

TypeError: Function.prototype.apply was called on undefined, which is a undefined and not a function

```
(let [blob (js/Blob. (clj->js ["export default function setup(sr) {console.log(1);}"]) (clj->js {:type "text/javascript"}))
      ;; Create a URL object from the Blob
      url (js/URL.createObjectURL blob)]
  (.then (js/import url)
      (fn [module] (println "work"))
      (fn [error] (println "error"))
  )
)
```

**Reply by borkdude** [reply]
Where are you executing this, in the scittle playground perhaps?

**Reply by borkdude** [reply]
Ah, `js/import` doesn't work by default in SCI since `import` is kind of a special thing in JS.
This is a workaround, add it to the top of your program:

```
(set! (.-import js/globalThis) (js/eval "(x) => import(x)"))
```

**Reply by borkdude** [reply]
The above should now work without workarounds in 0.622

**Comment by kangaroolab** [comment]
thanks!
